Use AttributeEventWait.waitForEvent() instead of deprecated AttributeEventWait.waitEvent()

reszelaz opened the pull request:
`AttributeEventWait.waitEvent()` use non-intuitive API and showed to be buggy - see [taurus-org/taurus#573](https://gitlab.com/taurus-org/taurus/-/issues/573) and
[taurus-org/taurus!1191](https://gitlab.com/taurus-org/taurus/-/merge_requests/1191).
Use an improved method `AttributeEventWait.waitForEvent()`.

Bump taurus version requirement.

Note, this PR is urgent to be reviewed and eventually merged since after the premature merge of https://github.com/sardana-org/sardana/pull/1592 currently the develop branch is in a broken state.

teresanunez reviewed (approved):
I don´t know how to test it, but it looks fine to me

reszelaz commented:
Thanks @teresanunez for the review. There are still some problems with the CI. Let's wait for the merge unless these are fixed.

The problem is when installing Sardana on the docker container for the tests:
```
┬─[zreszela@pt222:~]─[03:01:34 PM]
╰─>$ docker run -d --name=sardana-test -h sardana-test -e DISPLAY=$DISPLAY -e QT_X11_NO_MITSHM=1 -v /tmp/.X11-unix:/tmp/.X11-unix -v /homelocal/zreszela/workspace/sardana:/sardana -v /homelocal/zreszela/workspace/taurus:/taurus reszelaz/sardana-test               (base) 
19ab9a6ae65d9dad66822f925f7c3862486999b93e7cb3e64448a08abf9f4cbd
┬─[zreszela@pt222:~]─[03:01:40 PM]
╰─>$ docker exec -it sardana-test bash                                                                                                                                                                                                                                  (base) 
root@sardana-test:/# cd /sardana/
root@sardana-test:/sardana# pip3 install -e .
DEPRECATION: Python 3.5 reached the end of its life on September 13th, 2020. Please upgrade your Python as Python 3.5 is no longer maintained. pip 21.0 will drop support for Python 3.5 in January 2021. pip 21.0 will remove support for this functionality.
Obtaining file:///sardana
    ERROR: Command errored out with exit status 1:
     command: /usr/bin/python3 -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/sardana/setup.py'"'"'; __file__='"'"'/sardana/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /tmp/pip-pip-egg-info-hwcjh5d4
         cwd: /sardana/
    Complete output (23 lines):
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/sardana/setup.py", line 129, in <module>
        install_requires=install_requires
      File "/usr/lib/python3.5/distutils/core.py", line 108, in setup
        _setup_distribution = dist = klass(attrs)
      File "/usr/lib/python3/dist-packages/setuptools/dist.py", line 320, in __init__
        _Distribution.__init__(self, attrs)
      File "/usr/lib/python3.5/distutils/dist.py", line 253, in __init__
        getattr(self.metadata, "set_" + key)(val)
      File "/usr/lib/python3.5/distutils/dist.py", line 1208, in set_requires
        distutils.versionpredicate.VersionPredicate(v)
      File "/usr/lib/python3.5/distutils/versionpredicate.py", line 116, in __init__
        self.pred = [splitUp(aPred) for aPred in str.split(",")]
      File "/usr/lib/python3.5/distutils/versionpredicate.py", line 116, in <listcomp>
        self.pred = [splitUp(aPred) for aPred in str.split(",")]
      File "/usr/lib/python3.5/distutils/versionpredicate.py", line 26, in splitUp
        return (comp, distutils.version.StrictVersion(verStr))
      File "/usr/lib/python3.5/distutils/version.py", line 40, in __init__
        self.parse(vstring)
      File "/usr/lib/python3.5/distutils/version.py", line 137, in parse
        raise ValueError("invalid version number '%s'" % vstring)
    ValueError: invalid version number '4.9.0dev0'
    ----------------------------------------
WARNING: Discarding file:///sardana. Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.
ERROR: Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.
root@sardana-test:/sardana# pip3 --version 
pip 20.3.4 from /usr/local/lib/python3.5/dist-packages/pip (python 3.5)
root@sardana-test:/sardana# python3 -c "import taurus; print(taurus.core.release.version)"
4.9.0.dev0
root@sardana-test:/sardana#
```

reszelaz commented:
As a workaround I relaxed the Taurus requirement to `> 4.8.0`. This should work well for the moment. On Tuesday I'll try to make work the requirement `>= 4.9.0.dev1`. 
Merging as it is in order to fix the develop branch.

reszelaz commented:
> As a workaround I relaxed the Taurus requirement to `> 4.8.0`. This should work well for the moment. On Tuesday I'll try to make work the requirement `>= 4.9.0.dev1`.
> Merging as it is in order to fix the develop branch.

Should be fixed with dd8fb2f.
